src/libcsg/modules/io/pdbreader.cc

```diff
@@ -17,428 +17,460 @@
+
+        // This means only atm1 is attached to a molecule
+      } else if (mol_iter2 == atm_molecule.end()) {
+        // Add atm2 to the molecule that contains atm1
+        molecule_atms[mol_iter1->second].push_back(atm_id2);
+        // Associate atm1 with the molecule it is now part of
+        atm_molecule[atm_id2] = mol_iter1->second;
+
+      } else if (mol_iter1 != mol_iter2) {
+        // This means both atm1 and atm2 are attached to a molecule
+        // But if they are already attached to the same molecule there is
+        // nothing else to be done.
+        int chosen_mol;
+        int obsolete_mol;
+        // We will merge the atms to the molecule with the smallest index
+        if (mol_iter1->second < mol_iter2->second) {
+          chosen_mol = mol_iter1->second;
+          obsolete_mol = mol_iter2->second;
+        } else {
+          chosen_mol = mol_iter2->second;
+          obsolete_mol = mol_iter1->second;
+        }
+
+        // Now we will proceed to cycle through the atms that were in the now
+        // obsolete molecule and make sure they are pointing to the new molecule
+        for (auto atm_temp = molecule_atms[obsolete_mol].begin();
+             atm_temp != molecule_atms[obsolete_mol].end(); atm_temp++) {
 
-        int bond_indx = 0;
-        // Cyle through the bonds and add them to the appropriate molecule
-        for(auto bond_pair =bond_pairs.begin(); bond_pair!=bond_pairs.end(); bond_pair++){
-
-            int atm_id1 = bond_pair->at(0);
-            int atm_id2 = bond_pair->at(1);
-            // Should be able to just look at one of the atoms the bond is attached too
-            // because the other will also be attached to the same molecule. 
-            int mol_ind  = atm_molecule[atm_id1];
-            Molecule *mi = mol_map[mol_ind];      
-            // Grab the id of the bead associated with the atom
-            // It may be the case that the atom id's and bead id's are different
-            int bead_id1 = bead_vec.at(atm_id1-1)->getId();
-            int bead_id2 = bead_vec.at(atm_id2-1)->getId();
-            Interaction *ic = new IBond(bead_id1,bead_id2);
-            ic->setGroup("BONDS");
-            ic->setIndex(bond_indx);
-            bond_indx++;
-            ic->setMolecule(mol_reInd_map[mol_ind]);  
-            top.AddBondedInteraction(ic);
-            mi->AddInteraction(ic); 
+          atm_molecule[*atm_temp] = chosen_mol;
         }
-    
-        // Finally we want to build an exclusion matrix 
-        top.RebuildExclusions();
+
+        // Splicing will remove atoms from the now obsolete molecule and place
+        // them on the chosen molecule.
+        molecule_atms[chosen_mol].splice(molecule_atms[chosen_mol].end(),
+                                         molecule_atms[obsolete_mol]);
+
+        // Finally we will clear out the record of the obsolete molecule
+        molecule_atms.erase(obsolete_mol);
+      }
+    }
+#ifdef DEBUG
```

Comment:
Maybe instead of `DEBUG`, it is better to use 
```#ifndef NDEBUG```
@junghans, do you have any preference?

Reply:
Right, I guess that would be more cmake standard.

Reply:
Ok will do.